### PR TITLE
[feature](function) support `initcap` string function.

### PR DESCRIPTION
**Pull request by luozenglin**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/function_string.cpp [hunk 1]
```diff
@@ -265,6 +265,45 @@ struct TransferImpl {
...
+// Capitalize first letter
+struct NameToInitcap {
+    static constexpr auto name = "initcap";
+};
+
+struct InitcapImpl {
+    static Status vector(const ColumnString::Chars& data, const ColumnString::Offsets& offsets,
+                         ColumnString::Chars& res_data, ColumnString::Offsets& res_offsets) {
+        size_t offset_size = offsets.size();
+        res_offsets.resize(offsets.size());
```
Comment (HappenLee) [inline be/src/vec/functions/function_string.cpp:277]: here maybe need memcpy， the offset do not change

File: be/src/exprs/string_functions.cpp [hunk 2]
```diff
@@ -350,6 +350,29 @@ StringVal StringFunctions::upper(FunctionContext* context, const StringVal& str)
     return result;
 }
 
+StringVal StringFunctions::initcap(FunctionContext* context, const StringVal& str) {
+    if (str.is_null) {
+        return StringVal::null();
+    }
+    StringVal result(context, str.len);
+    if (UNLIKELY(result.is_null)) {
```
[the remark this answers is not in the document]
Comment (luozenglin) [reply be/src/exprs/string_functions.cpp]: I have removed it. Thanks.

File: be/src/vec/functions/function_string.cpp (hunk 1 above)
Comment (luozenglin) [reply be/src/vec/functions/function_string.cpp:277]: done.

**Review by HappenLee — DISMISSED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**